app/controllers/favorites_controller.rb

```diff
@@ -11,11 +11,17 @@ def update
     redirect_to "/pets/#{params[:pet_id]}"
   end
 
-  def destroy
+  def delete
     pet = Pet.find(params[:pet_id])
     favorite.remove_fav(params[:pet_id])
     session[:favorite] = favorite.contents
     flash[:notice] = "Removed #{pet.name} from favorites."
     redirect_to params[:prev_page]
   end
+
+  def destroy
+    favorite.remove_all
+    session[:favorite] = favorite.contents
```

Comment:
I couldn't agree more with you, these are similar but different enough to be two individual methods in the same class.